static/templates/single_message.hbs

```diff
@@ -1,5 +1,5 @@
 <div zid="{{msg/id}}" id="{{table_name}}{{msg/id}}"
-  class="message_row{{^msg/is_stream}} private-message{{/msg/is_stream}}{{#include_sender}} include-sender{{/include_sender}}{{#contains_mention}} mention{{/contains_mention}}{{#include_footer}} last_message{{/include_footer}}{{#msg.unread}} unread{{/msg.unread}} {{#if msg.locally_echoed}}local{{/if}} selectable_row"
+  class="message_row{{^msg/is_stream}} private-message{{/msg/is_stream}}{{#include_sender}} include-sender{{/include_sender}}{{#contains_direct_user_mention}} direct_mention{{/contains_direct_user_mention}}{{#contains_group_mention}} group_mention{{/contains_group_mention}}{{#include_footer}} last_message{{/include_footer}}{{#msg.unread}} unread{{/msg.unread}} {{#if msg.locally_echoed}}local{{/if}} selectable_row"
```

Comment:
As a small comment on commit structure -- I think you could extract a prep commit that just adds the separate classes for direct/group mentions (and logic for that) without any visible styling changes, and we could merge that now, which would leave just a small CSS change to do the coloring remaining.

Reply:
Just want to note that I personally in the past have avoided merging commits in isolation that add new things that aren't used yet, since until the thing that uses it is merged I don't know for sure if I'll want to change the setup in some way (and it's also sometimes harder to look back at commits and understand why that change was made). I trust that this has worked fairly well at Zulip, so I'm happy to do it, but just wanted to share why I haven't been doing it that much.

Reply:
Yeah, I think it's pretty clearly useful to be able to distinguish group/direct mentions, and we'll end up using the information somehow in the design, so I think it's pretty safe in this case.